### ISSUE TITLE
 at android.view.ViewGroup.getAndVerifyPreorderedView(ViewGroup.java:3756)

### ISSUE THREAD

**Issue by moezisar** [burhanrashid52/PhotoEditor#128]
Image not getting saved. 
java.lang.IndexOutOfBoundsException: Index: 1, Size: 0

 drawingCache = parentView.getDrawingCache();

Exception at this line drawing cache null.

**Comment by ankitivo** [comment]
I am also experiencing the same. Here is the full stack trace reported through firebase:
```
Fatal Exception: java.lang.IndexOutOfBoundsException: Index: 1, Size: 0
       at java.util.ArrayList.get(ArrayList.java:437)
       at android.view.ViewGroup.getAndVerifyPreorderedView(ViewGroup.java:3634)
       at android.view.ViewGroup.gatherTransparentRegion(ViewGroup.java:7087)
       at android.view.ViewGroup.gatherTransparentRegion(ViewGroup.java:7089)
       at android.view.ViewGroup.gatherTransparentRegion(ViewGroup.java:7089)
       at android.view.ViewGroup.gatherTransparentRegion(ViewGroup.java:7089)
       at android.view.ViewGroup.gatherTransparentRegion(ViewGroup.java:7089)
       at android.view.ViewGroup.gatherTransparentRegion(ViewGroup.java:7089)
       at android.view.ViewGroup.gatherTransparentRegion(ViewGroup.java:7089)
       at android.view.ViewGroup.gatherTransparentRegion(ViewGroup.java:7089)
       at android.view.ViewGroup.gatherTransparentRegion(ViewGroup.java:7089)
       at android.view.ViewGroup.gatherTransparentRegion(ViewGroup.java:7089)
       at com.android.internal.policy.DecorView.gatherTransparentRegion(DecorView.java:306)
       at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2355)
       at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1482)
       at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7209)
       at android.view.Choreographer$CallbackRecord.run(Choreographer.java:949)
       at android.view.Choreographer.doCallbacks(Choreographer.java:761)
       at android.view.Choreographer.doFrame(Choreographer.java:696)
       at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:935)
       at android.os.Handler.handleCallback(Handler.java:873)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loop(Looper.java:193)
       at android.app.ActivityThread.main(ActivityThread.java:6865)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:504)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

From firebase analytics, I could see that the user selected a bunch of filters and finally a heart-shaped emoticon before trying to save the edited image and run into this issue.

LMK, if the name of the applied filters and the emoticon could be of importance here (I'll have to reconcile the list).

**Comment by ankitbatra11** [comment]
I am also experiencing the same. Here is the full stack trace reported through firebase:
```
Fatal Exception: java.lang.IndexOutOfBoundsException: Index: 1, Size: 0
       at java.util.ArrayList.get(ArrayList.java:437)
       at android.view.ViewGroup.getAndVerifyPreorderedView(ViewGroup.java:3634)
       at android.view.ViewGroup.gatherTransparentRegion(ViewGroup.java:7087)
       at android.view.ViewGroup.gatherTransparentRegion(ViewGroup.java:7089)
       at android.view.ViewGroup.gatherTransparentRegion(ViewGroup.java:7089)
       at android.view.ViewGroup.gatherTransparentRegion(ViewGroup.java:7089)
       at android.view.ViewGroup.gatherTransparentRegion(ViewGroup.java:7089)
       at android.view.ViewGroup.gatherTransparentRegion(ViewGroup.java:7089)
       at android.view.ViewGroup.gatherTransparentRegion(ViewGroup.java:7089)
       at android.view.ViewGroup.gatherTransparentRegion(ViewGroup.java:7089)
       at android.view.ViewGroup.gatherTransparentRegion(ViewGroup.java:7089)
       at android.view.ViewGroup.gatherTransparentRegion(ViewGroup.java:7089)
       at com.android.internal.policy.DecorView.gatherTransparentRegion(DecorView.java:306)
       at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2355)
       at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1482)
       at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7209)
       at android.view.Choreographer$CallbackRecord.run(Choreographer.java:949)
       at android.view.Choreographer.doCallbacks(Choreographer.java:761)
       at android.view.Choreographer.doFrame(Choreographer.java:696)
       at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:935)
       at android.os.Handler.handleCallback(Handler.java:873)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loop(Looper.java:193)
       at android.app.ActivityThread.main(ActivityThread.java:6865)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:504)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

From firebase analytics, I could see that the user selected a bunch of filters and finally a heart-shaped emoticon before trying to save the edited image and run into this issue.

LMK, if the name of the applied filters and the emoticon could be of importance here (I'll have to reconcile the list).

**Comment by burhanrashid52** [comment]
This issue is fixed in 1.1.0 release. Reopen this if you are still facing this issue.
